app/api/items/save_grade.go

```diff
@@ -103,18 +103,31 @@ func saveGradingResultsIntoDB(store *database.DataStore, user *database.User,
 				END, score_computed, 0), 100)`, score, score)
 	values := []interface{}{
 		1,
-		database.Now(),
+		// store the old value into a temporary variable
+		gorm.Expr("(@old_latest_answer_at:=latest_answer_at)"),
```

Comment:
why don't you do a "simple" JOIN with the UPDATE?

Reply:
fixed